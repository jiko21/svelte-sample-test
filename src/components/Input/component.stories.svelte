<script>
  import { Meta, Template, Story } from '@storybook/addon-svelte-csf';
  import Input from './component.svelte';
</script>

<Meta
  title="components/Input"
  component={Input}
  argTypes={{
    type: {
      control: 'select',
      options: ['text', 'email', 'password'],
    },
  }}
/>

<Template let:args>
  <Input {...args} />
</Template>

<Story
  name="text"
  args={{
    type: 'text',
  }}
/>

<Story
  name="email"
  args={{
    type: 'email',
  }}
/>

<Story
  name="password"
  args={{
    type: 'password',
  }}
/>