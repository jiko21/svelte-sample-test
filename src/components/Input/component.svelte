<script lang="ts">
  export let type, value, testId;
  const typeAction = (node) => {
    node.type = type;
  }
</script>
<input use:typeAction bind:value data-testid={testId} />
<style>
  input {
    box-sizing: border-box;
    border: solid 1px #cccccc;
    border-radius: 4px;
    line-height: 16px;
    padding: 8px 4px;
    height: 32px;
  }
</style>