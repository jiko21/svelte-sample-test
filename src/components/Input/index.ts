import Input from './component.svelte';

export default Input;
