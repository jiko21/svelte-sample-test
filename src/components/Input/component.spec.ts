import { fireEvent, screen } from '@testing-library/dom';
import {render} from '@testing-library/svelte'

import Input from './component.svelte';

describe('components/Input', () => {
  describe('primary', () => {
    it('correctly render', () => {
      const component = render(Input);
      expect(component.container).toMatchSnapshot();
    });

    it('call function correctly', () => {
      render(Input, {
        testId: 'testId',
      });
      const input: any = screen.getByTestId('testId');
      fireEvent.change(input, {
        target: {
          value: 'test'
        }
      })
      expect(input.value).toBe('test');
    });
  });
})
