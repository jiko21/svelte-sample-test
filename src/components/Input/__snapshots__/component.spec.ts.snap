// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input primary correctly render 1`] = `
<body>
  <div>
    <input
      class="svelte-9hrj2z"
      type="undefined"
    />
  </div>
</body>
`;
